src/s11/handlers/s11_CB_req_handler.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2020, Infosys Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+#include <string.h>
+#include <pthread.h>
+
+#include "err_codes.h"
+#include "s11_options.h"
+#include "ipc_api.h"
+#include "s11.h"
+#include "s11_config.h"
+#include "msgType.h"
+
+#include <gtpV2StackWrappers.h>
+/*Globals and externs*/
+extern int g_resp_fd;
+extern struct GtpV2Stack* gtpStack_gp;
+/*End : globals and externs*/
+
+uint64_t conv_uint8_arr_to_uint64(const uint8_t* bit_rate_array)
+{
+    uint64_t bit_rate_kbps = (*(bit_rate_array) << 32) | (*(bit_rate_array+1) << 24) | (*(bit_rate_array+2) << 16) | 
+	    			(*(bit_rate_array+3) << 8) | (*(bit_rate_array+4));
+    return bit_rate_kbps;
+}
+
+
+void build_bearer_qos(BearerQosIeData* s11_bearer_qos, bearer_qos_t* mme_bearer_qos)
+{
+	mme_bearer_qos->arp.prioLevel = s11_bearer_qos->pl;
+	mme_bearer_qos->arp.preEmptionCapab = s11_bearer_qos->pci;
+	mme_bearer_qos->arp.preEmptionVulnebility = s11_bearer_qos->pvi;
+
+	mme_bearer_qos->qci = s11_bearer_qos->qci;
+	
+	mme_bearer_qos->mbr_ul = conv_uint8_arr_to_uint64(s11_bearer_qos->maxBitRateUl.values);
+	mme_bearer_qos->mbr_dl = conv_uint8_arr_to_uint64(s11_bearer_qos->maxBitRateDl.values);
+	mme_bearer_qos->gbr_ul = conv_uint8_arr_to_uint64(s11_bearer_qos->guraranteedBitRateUl.values);
+	mme_bearer_qos->gbr_dl = conv_uint8_arr_to_uint64(s11_bearer_qos->guaranteedBitRateDl.values);
+}
+
+int
+s11_CB_req_handler(MsgBuffer* message, GtpV2MessageHeader* hdr, uint32_t sgw_ip)
+{
+
+	struct cb_req_Q_msg cbr_info;
+
+	cbr_info.header.ue_idx = hdr->teid;
+	cbr_info.header.msg_type = create_bearer_request;
```

Comment:
should store the seq number in cbr_info to be used to send back cb response.

Reply:
Taken care.